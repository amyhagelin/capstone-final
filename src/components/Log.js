import React, { Component } from "react";
import {connect} from 'react-redux';
import SingleEventComponent from './Event';
import { getEvents } from '../actions/getEvents';

export class Log extends Component {
	static defaultProps = {
		eventLog: []
	}
	
	componentDidMount () {
		this.props.dispatch(getEvents());
		console.log('dispatch get events from log')
		console.log(this.props.eventLog)

	}

	render() {
		return (
			<div>
				<h2>Seizure List</h2>
				{ 
					this.props.eventLog.map((singleEvent) => {
						return <SingleEventComponent key={ singleEvent._id } event={ singleEvent } />
					}) 
				}	
			</div>
		);
	}
}


const mapStateToProps = state => ({
    eventLog: state.event.log
});

export default connect(mapStateToProps)(Log);
