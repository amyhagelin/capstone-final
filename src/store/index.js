// make a component and connect to store
// component can be a button w/ a click listener, dispatch the login action


import {createStore, applyMiddleware} from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { routerMiddleware } from 'react-router-redux';
import createHistory from 'history/createBrowserHistory';
import thunk from 'redux-thunk';

// import auth from '../reducers/auth'
import combinedReducers from '../reducers/index';

export const history = createHistory();

export default createStore(
    combinedReducers, 
    composeWithDevTools(), 
    applyMiddleware(thunk, routerMiddleware(history))
);