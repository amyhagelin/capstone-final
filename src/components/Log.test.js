import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import SingleEventComponent from './Event';
import { MemoryRouter } from 'react-router'
import {history } from '../store'
import { Log } from './Log';
import { BrowserRouter as Router } from 'react-router-dom';

describe ('<Log />', () => {

    it('renders correctly', () => {
        const eventLog = [{
            _id: 0,
            date: new Date(123)
        }, {
            _id: 1,
            date: new Date(123)
        }]
        const dispatch = jest.fn()
        const tree = renderer.create(
            <Router>
                <Log eventLog={ eventLog } dispatch={ dispatch } />
            </Router>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Renders with proper data', () => {
        const dispatch = jest.fn()
        const wrapper = mount(<Log 
            dispatch={ dispatch } 
            eventLog={ [{ _id: 1 }] }
        />);

        console.log(dispatch.mock.calls[0][0]);

        expect(wrapper.find(SingleEventComponent).length).toEqual(1)
    });

    it('Fires the correct dispatch', () => {
        const dispatch = jest.fn();
        const wrapper = mount(<Log dispatch={ dispatch }/>);
        expect(dispatch).toHaveBeenCalled();
    });

    it('Renders the single events', () => {
        const eventLog = [{
            _id: 0,
            date: new Date(123)
        }, {
            _id: 1,
            date: new Date(123)
        }]
        const wrapper = shallow(<Log eventLog={eventLog} />);
        const eventComponents = wrapper.find(SingleEventComponent);
        expect(eventComponents.length).toEqual(eventLog.length);

    });

});