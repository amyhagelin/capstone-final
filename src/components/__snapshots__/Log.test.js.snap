// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Log /> renders correctly 1`] = `
<div>
  <h2>
    Seizure List
  </h2>
  <div
    className="single-event"
  >
    January 1, 1970
     | 
     | 
     | 
    <br />
    Triggers: 
     
    <br />
    Medication: 
     
    <br />
    Notes: 
  </div>
  <div
    className="single-event"
  >
    January 1, 1970
     | 
     | 
     | 
    <br />
    Triggers: 
     
    <br />
    Medication: 
     
    <br />
    Notes: 
  </div>
</div>
`;
